caikit/core/modules/base.py

```diff
@@ -79,6 +79,17 @@ def metadata(self) -> Dict[str, Any]:
             self._metadata = {}
         return self._metadata
 
+    def get_metadata(self) -> Dict[str, str]:
```

Comment:
After @joerunde pointed out that we already have `def metadata` above, I think we should figure out how to combine these two. My thought would be that this addition of `"name"` and `"version"` could/should be added by the service on top of the output of the `metadata` property.